share/spack/qa/setup-env-test.fish

```diff
@@ -116,7 +118,7 @@ function spt_fails
 
     if test $status -eq 0
         fail
-        echo_red "Command failed with error $status"
+        echo_red "Command succeeded, but should fail"
```

Comment:
It's not clear to me why the change above is not needed here.

Reply:
In this case $status is used only once, in the others it's used twice, and then the second time it refers to the status of another command

Reply:
In this case it doesn't make sense to print the status code, since it's testing if the command fails to fail, so it's always 0 anyways.

Reply:
Oh, I missed that it's not in the string anymore. Can we use `cmd_status` also here, for better clarity if we ever change this file, and add a comment in both cases above the line where we:
```
set cmd_status $status
```
that says we record the status since following command may override it? Other than this minor things this LGTM.

Reply:
Hm, but the status in `spt_fails` is not really interesting, if it fails it can only be 0.